File: Python/Program-Timer/timer.py
import subprocess
import time
import sys

# Check if user has provided commandline arguments or not.
if len(sys.argv) > 1:
    target_prog_params = sys.argv[1:]
else:
    print("Target program not specified...")

# Generate the command to run the script from commandline.
command = ["python"] + target_prog_params

print("Target Output:\n")

# Calculate the time taken to execute the script.
initial_time = time.time()
status = subprocess.call(command)
time_taken = round(time.time() - initial_time, 3)

# Print the time taken to execute the script and the user know if the execution was
# successful or not.
if status == 0:
    print("\033[92m"+f"\n{sys.argv[1]} took {time_taken} seconds to execute successfully."+"\033[0m")
else:
    print("\033[91m"+f"\n{sys.argv[1]} took {time_taken} seconds before failing."+"\033[0m")

File: C/Program-Timer/timer.c
#include<time.h>
#include<stdio.h>

// Placeholer function to edit:
void function(int a, int b){
    printf("%d + %d = %d\n", a, b, a+b);
}

int main(int argc, char *argv[])
{
    /*
    Create start and end clock variables of type `clock_t`
    and time the function.
    */
    clock_t start = clock();
    function(23, 54);
    clock_t end = clock();

    // Divide clock difference with CLOCKS_PER_SEC to convert time to seconds
    // and then typecast it to double.
    double time_taken = ((double)(end - start))/CLOCKS_PER_SEC;

    printf("function took %f seconds to execute\n", time_taken);

    return 0;
}
